zerver/views/upload.py

```diff
@@ -82,7 +83,7 @@ def serve_file(request: HttpRequest, user_profile: UserProfile,
 
     return serve_s3(request, path_id, url_only)
 
-def serve_local_file_unauthed(request: HttpRequest, token: str) -> HttpResponse:
+def serve_local_file_unauthed(request: HttpRequest, token: str, *args: Any) -> HttpResponse:
```

Comment:
I think we should call that second argument `filename`, and validate the file's basename in this code path matches what's at the end of the `path_id`.  It seems safer to not allow clients to modify the end of these URLs freely and still get the same file.

@andersk FYI.

Reply:
Does the browser actually care about what's in the url? Doesn't it look at the response headers we set to the real filename anyway? But agreed, makes sense to have that additional sanity check either way, added it